libraries/Microsoft.Bot.Builder.LUIS/LuisRecognizer.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Cognitive.LUIS;
+using Microsoft.Cognitive.LUIS.Models;
+using Newtonsoft.Json.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Bot.Builder.LUIS
+{
+    /// <inheritdoc />
+    /// <summary>
+    /// A Luis based implementation of IRecognizer
+    /// </summary>
+    public class LuisRecognizer : IRecognizer
+    {
+        private readonly LuisService _luisService;
+        private readonly ILuisOptions _luisOptions;
+        private readonly ILuisRecognizerOptions _luisRecognizerOptions;
+
+        public LuisRecognizer(ILuisModel luisModel, ILuisRecognizerOptions luisRecognizerOptions = null, ILuisOptions options = null)
+        {
+            _luisService = new LuisService(luisModel);
+            _luisOptions = options;
+            _luisRecognizerOptions = luisRecognizerOptions;
+        }
+
+        /// <inheritdoc />
+        public Task<RecognizerResult> Recognize(string utterance, CancellationToken ct)
+        {
+            var luisRequest = new LuisRequest(utterance);
+            _luisOptions?.Apply(luisRequest);
+            return Recognize(luisRequest, ct, _luisRecognizerOptions?.Verbose ?? true);
```

Comment:
Special cases like this will go away if you use the default option approach. That'll also get rid of all the null-coalescing operators which make code hard to write, less clear, and more bug prone.

Reply:
Done